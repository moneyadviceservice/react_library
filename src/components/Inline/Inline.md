### A modified `Paragraph` component that renders as an inline `span` element using its inherited styles.
The `Inline` component is used for single line text. The component renders a `<span>` element by default. If you need a multiline paragraph use the `Paragraph` component. You can use all the props available to the `Paragraph` component here too!

```jsx
import { Col, Inline } from '@moneypensionservice/directories';

<>
<Col>
  <Inline strike>The quick brown fox jumps over the lazy dog</Inline>
</Col>
<Col>
  <Inline>The quick brown fox jumps over the lazy dog</Inline>
</Col>
<Col>
  <Inline color='salmon'>The quick brown fox jumps over the lazy dog</Inline>
</Col>
</>
```
