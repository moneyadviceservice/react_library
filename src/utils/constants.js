export const breakpoints = {
  xs: 1,
  sm: 48, // 768px
  md: 64, // 1024px
  lg: 90, // 1440px
  xl: 120, // 1920px
}

export const dimensions = ['xs', 'sm', 'md', 'lg', 'xl']

export const mediaQuery = 'only screen'

export const ALIGN_SELF_MAP = {
  center: 'center',
  end: 'flex-end',
  start: 'flex-start',
  stretch: 'stretch',
}

export const SPACING_SIZES = [
  'xxsmall',
  'xsmall',
  'small',
  'medium',
  'large',
  'xlarge',
]

