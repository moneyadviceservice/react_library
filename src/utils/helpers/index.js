export * from './background'
export * from './responsive'
export * from './flexbox'
export * from './styles'
export * from './mediaQueries'
export * from './sizing'
export * from './refs'
