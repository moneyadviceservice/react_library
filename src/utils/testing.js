import { render, act } from '@testing-library/react'
import GlobalTheme from '../components/ThemeProvider'

const customRender = (ui, options) =>
  render(ui, { wrapper: GlobalTheme, ...options })

const waitRender = async (ui, options) => {
  const { container, ...rest } = customRender(ui, options)

  await act(async () => {
    await Promise.resolve(container)
  })

  return { container, ...rest }
}

export * from '@testing-library/react'

export { customRender as render, waitRender }
