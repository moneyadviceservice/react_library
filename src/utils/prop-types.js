import PropTypes from 'prop-types'
import { SPACING_SIZES } from './constants'

/** spacing prop.type */
const spacingProp = PropTypes.shape({
  bottom: PropTypes.oneOfType([
    PropTypes.oneOf(SPACING_SIZES),
    PropTypes.string,
  ]),
  horizontal: PropTypes.oneOfType([
    PropTypes.oneOf(SPACING_SIZES),
    PropTypes.string,
  ]),
  left: PropTypes.oneOfType([PropTypes.oneOf(SPACING_SIZES), PropTypes.string]),
  right: PropTypes.oneOfType([
    PropTypes.oneOf(SPACING_SIZES),
    PropTypes.string,
  ]),
  top: PropTypes.oneOfType([PropTypes.oneOf(SPACING_SIZES), PropTypes.string]),
  vertical: PropTypes.oneOfType([
    PropTypes.oneOf(SPACING_SIZES),
    PropTypes.string,
  ]),
})

export const genericPropTypes = {
  a11yTitle: PropTypes.string,
  alignSelf: PropTypes.oneOf(['start', 'center', 'end', 'stretch']),
  margin: PropTypes.oneOfType([
    PropTypes.oneOf(['none', ...SPACING_SIZES]),
    spacingProp,
    PropTypes.string,
    PropTypes.object,
  ]),
  padding: PropTypes.oneOfType([
    PropTypes.oneOf([...SPACING_SIZES]),
    spacingProp,
    PropTypes.string,
    PropTypes.object,
  ]),
  border: PropTypes.oneOfType([
    PropTypes.oneOf(['none', ...SPACING_SIZES]),
    spacingProp,
    PropTypes.string,
    PropTypes.object,
  ]),
}

export function genericPropsDefaults({
  a11yTitle = null,
  alignSelf = null,
  margin = null,
  padding = null,
  border = null,
} = {}) {
  return {
    a11yTitle: a11yTitle,
    alignSelf: alignSelf,
    margin: margin,
    padding: padding,
    border: border,
  }
}
