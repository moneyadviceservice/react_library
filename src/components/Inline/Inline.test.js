import React from 'react'
import { render } from '../../utils/testing'
import { Inline } from '../Inline'

describe('<Inline />', () => {
  describe('Snapshots', () => {
    it('should match snapshot with children element', () => {
      const { container } = render(<Inline>example child</Inline>)
      expect(container).toMatchSnapshot()
    })
  })
})
