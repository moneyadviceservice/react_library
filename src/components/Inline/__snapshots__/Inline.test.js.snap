// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Inline /> Snapshots should match snapshot with children element 1`] = `
<div>
  <span
    class="StyledParagraph-sc-1toaej-0 gpltyO"
  >
    example child
  </span>
</div>
`;
